cmd/agent/api/server_cmd.go

```diff
@@ -0,0 +1,159 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package api
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"fmt"
+	"net"
+	"net/http"
+	"time"
+
+	gorilla "github.com/gorilla/mux"
+	grpc_auth "github.com/grpc-ecosystem/go-grpc-middleware/auth"
+	"github.com/grpc-ecosystem/grpc-gateway/runtime"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+
+	"github.com/DataDog/datadog-agent/cmd/agent/api/internal/agent"
+	"github.com/DataDog/datadog-agent/cmd/agent/api/internal/check"
+	"github.com/DataDog/datadog-agent/comp/aggregator/demultiplexer"
+	"github.com/DataDog/datadog-agent/comp/core/flare"
+	"github.com/DataDog/datadog-agent/comp/core/secrets"
+	"github.com/DataDog/datadog-agent/comp/core/workloadmeta"
+	workloadmetaServer "github.com/DataDog/datadog-agent/comp/core/workloadmeta/server"
+	"github.com/DataDog/datadog-agent/comp/dogstatsd/replay"
+	dogstatsdServer "github.com/DataDog/datadog-agent/comp/dogstatsd/server"
+	dogstatsddebug "github.com/DataDog/datadog-agent/comp/dogstatsd/serverDebug"
+	logsAgent "github.com/DataDog/datadog-agent/comp/logs/agent"
+	"github.com/DataDog/datadog-agent/comp/metadata/host"
+	"github.com/DataDog/datadog-agent/comp/metadata/inventoryagent"
+	"github.com/DataDog/datadog-agent/comp/metadata/inventoryhost"
+	"github.com/DataDog/datadog-agent/pkg/aggregator/sender"
+	"github.com/DataDog/datadog-agent/pkg/config"
+	remoteconfig "github.com/DataDog/datadog-agent/pkg/config/remote/service"
+	pb "github.com/DataDog/datadog-agent/pkg/proto/pbgo/core"
+	"github.com/DataDog/datadog-agent/pkg/tagger"
+	taggerserver "github.com/DataDog/datadog-agent/pkg/tagger/server"
+	grpcutil "github.com/DataDog/datadog-agent/pkg/util/grpc"
+	"github.com/DataDog/datadog-agent/pkg/util/optional"
+)
+
+var cmdListener net.Listener
+
+func startCMDServer(
+	cmdAddr string,
+	tlsConfig *tls.Config,
+	tlsCertPool *x509.CertPool,
+	configService *remoteconfig.Service,
+	flare flare.Component,
+	dogstatsdServer dogstatsdServer.Component,
+	capture replay.Component,
+	serverDebug dogstatsddebug.Component,
+	wmeta workloadmeta.Component,
+	logsAgent optional.Option[logsAgent.Component],
+	senderManager sender.DiagnoseSenderManager,
+	hostMetadata host.Component,
+	invAgent inventoryagent.Component,
+	demux demultiplexer.Component,
+	invHost inventoryhost.Component,
+	secretResolver secrets.Component,
+) (err error) {
+	// get the transport we're going to use under HTTP
+	cmdListener, err = getListener(cmdAddr)
+	if err != nil {
+		// we use the listener to handle commands for the Agent, there's
+		// no way we can recover from this error
+		return fmt.Errorf("Unable to create the api server: %v", err)
```

Comment:
I would maybe modify the error messages in the copied code to indicate that this is for the CMD API server.